include/ofi_list.h

```diff
@@ -579,6 +579,35 @@ slist_splice_tail(struct slist *dst, struct slist *src)
 	return dst;
 }
 
+typedef void (*slist_swap_func)(struct slist_entry *entry1, struct slist_entry *entry2);
+typedef int (*slist_compare_func)(struct slist_entry *entry1, struct slist_entry *entry2);
+
+static inline void
+slist_sort(struct slist *list, slist_swap_func swap, slist_compare_func comp)
+{
+	int swapped;
+	struct slist_entry *cur_ptr;
+	struct slist_entry *end_ptr = NULL;
+
+	if (slist_empty(list))
+		return;
+
+	do {
+		swapped = 0;
+		cur_ptr = list->head;
+  
+		while (cur_ptr->next != end_ptr) {
+			if (comp(cur_ptr, cur_ptr->next)) {
+				swap(cur_ptr, cur_ptr->next);
+				swapped = 1;
+			}
+			cur_ptr = cur_ptr->next;
+		}
+
+		end_ptr = cur_ptr;
+	} while (swapped);
+}
```

Comment:
Can this be replaced with inserting at a sorted location instead of sorting afterwards?  I.e. use slist_insert_before_first_match?

Reply:
Good catch! Done